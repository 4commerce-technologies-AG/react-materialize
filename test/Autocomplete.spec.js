import React from 'react';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';
import Autocomplete from '../src/Autocomplete';
import mocker from './helper/new-mocker';

describe('<Autocomplete />', () => {
  const autocompleteInitMock = jest.fn();
  const data = {
    Apple: null,
    Microsoft: null,
    Google: 'http://placehold.it/250x250',
    'Apple Inc': null
  };
  const componentId = 'testAutocompleteId';

  let wrapper;

  beforeEach(() => {
    autocompleteInitMock.mockClear();
  });

  test('renders', () => {
    const { container } = render(
      <Autocomplete title="Test Title" id={componentId} options={{ data }} />
    );

    expect(container).toMatchSnapshot();
  });

  test('handles layout classes', () => {
    const { container } = render(<Autocomplete s={4} m={6} />);

    expect(container).toMatchSnapshot();
  });

  test('renders placeholder', () => {
    const { getByPlaceholderText } = render(
      <Autocomplete placeholder="Name" />
    );

    expect(getByPlaceholderText('Name')).toBeTruthy();
  });

  describe('initialises', () => {
    const autocompleteInstanceDestroyMock = jest.fn();
    const autocompleteMock = {
      init: (_, options) => {
        autocompleteInitMock(options);
        return {
          destroy: autocompleteInstanceDestroyMock
        };
      }
    };

    mocker('Autocomplete', autocompleteMock);

    beforeEach(() => {
      autocompleteInitMock.mockClear();
      autocompleteInstanceDestroyMock.mockClear();
    });

    test('calls Autocomplete', () => {
      render(<Autocomplete />);
      expect(autocompleteInitMock).lastCalledWith({
        data: {},
        limit: Infinity,
        minLength: 1,
        onAutocomplete: null,
        sortFunction: null
      });
    });
  });

  describe('with new data', () => {
    test('reinitializes', () => {
      const { rerender } = render(<Autocomplete />);

      rerender(<Autocomplete options={{ data: { foo: 'bar' } }} />);

      expect(autocompleteInitMock).lastCalledWith({
        data: {
          foo: 'bar'
        }
      });
    });
  });
});
