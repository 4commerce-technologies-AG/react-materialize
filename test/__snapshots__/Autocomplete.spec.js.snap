// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Autocomplete /> handles layout classes 1`] = `
<div>
  <div
    class="input-field col s4 m6"
  >
    <input
      class="autocomplete"
      id="Autocomplete-mockId"
      type="text"
    />
    <label
      for="Autocomplete-mockId"
    />
  </div>
</div>
`;

exports[`<Autocomplete /> renders 1`] = `
<div>
  <div
    class="input-field col"
  >
    <input
      class="autocomplete"
      id="testAutocompleteId"
      type="text"
    />
    <label
      for="testAutocompleteId"
    >
      Test Title
    </label>
  </div>
</div>
`;
